common-pages/src/TimeWindowUtils.tsx

```diff
@@ -0,0 +1,106 @@
+﻿//******************************************************************************************************
+//  TimeWindowUtils.tsx - Gbtc
+//
+//  Copyright © 2023, Grid Protection Alliance.  All Rights Reserved.
+//
+//  Licensed to the Grid Protection Alliance (GPA) under one or more contributor license agreements. See
+//  the NOTICE file distributed with this work for additional information regarding copyright ownership.
+//  The GPA licenses this file to you under the MIT License (MIT), the "License"; you may not use this
+//  file except in compliance with the License. You may obtain a copy of the License at:
+//
+//      http://opensource.org/licenses/MIT
+//
+//  Unless agreed to in writing, the subject software distributed under the License is distributed on an
+//  "AS-IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. Refer to the
+//  License for the specific language governing permissions and limitations.
+//
+//  Code Modification History:
+//  ----------------------------------------------------------------------------------------------------
+//  07/11/2023 - C. Lackner
+//       Generated original version of source code.
+//******************************************************************************************************
+import moment from 'moment';
+
+/**
+ * Description
+ */
+export interface ITimeFilter {
```

Comment:
typing is not great.
If `ITimeFilter` is a prop to `TimeFilter` it should not require an overconstraint (start,end,center and duration)
You should define a `ITimeWindow` interface that could be start/end or start/duration or center/duration (half).....